server/tests/routes/experiments/biomed_nl/traversal_test.py

```diff
@@ -0,0 +1,572 @@
+              'name': {
+                  'value': 'genvar assoc name'
+              }
+          }
+      }
+  }
+
+  return {dcid: triples.get(dcid, {}).get(out, {}) for dcid in dcids}
+
+
+class TestTraversal(unittest.TestCase):
+
+  @patch('server.lib.fetch.triples')
+  def test_get_next_hop_triples(self, mock_fetch_triples):
+    # Mock the return value of fetch.triples
+    mock_fetch_triples.side_effect = fetch_triples_side_effect
+
+    outgoing_result = get_next_hop_triples(['gene1', 'disease1'], out=True)
+
+    mock_fetch_triples.assert_called_once()
+
+    # Assertions
+    expected_outgoing_result = {
+        "gene1": {
+            "(ortholog)": {"gene2"},
+            "(symbol)": set(),
+        },  # prop2 has a literal
+        "disease1": {
+            "(symptom)": {"symptom1", 'symptom2'},
+            '(severity)': set()
+        }  # prop3 is skipped because its a terminal
+    }
+
+    assert DeepDiff(outgoing_result,
+                    expected_outgoing_result,
+                    ignore_order=True) == {}
+
+    incoming_result = get_next_hop_triples(['gene1', 'disease1'], out=False)
+    expected_incoming_result = {
+        "gene1": {
+            "(DiseaseGeneAssociations linked by geneId)": {'gene_disease1'}
+        },  # prop2 has a literal
+        "disease1": {
+            '(DiseaseGeneAssociations linked by diseaseId)': {
+                'gene_disease1', 'gene_disease2'
+            },
+            '(GeneticVariantDiseaseAssociations linked by diseaseId)': {
+                'genetic_variant_disease1'
+            }
+        }  # prop3 is skipped because its a terminal
+    }
+    assert DeepDiff(incoming_result,
+                    expected_incoming_result,
+                    ignore_order=True) == {}
+
+  def test_merge_triples(self):
+    path_store = PathStore()
+    path_store.path_store = {
+        'dcid1': {
+            '(prop1) (prop2)': {'dcid3'},
+            '(prop3)': {'dcid4'}
+        },
+        'dcid2': {
+            '(prop4) (prop5)': {'dcid5'}
+        }
+    }
+
+    input_triples = {
+        'dcid3': {
+            '(propA)': {'dcid6'},
+            '(propB)': {'dcid7'}
+        },
+        'dcid4': {
+            '(propC)': {'dcid8'}
+        },
+        'dcid5': {
+            '(propD)': {'dcid9'}
+        }
+    }
+
+    path_store.merge_triples_into_path_store(input_triples)
+
+    expected_path_store = {
+        'dcid1': {
+            '(prop1) (prop2) (propA)': {'dcid6'},
+            '(prop1) (prop2) (propB)': {'dcid7'},
+            '(prop3) (propC)': {'dcid8'}
+        },
+        'dcid2': {
+            '(prop4) (prop5) (propD)': {'dcid9'}
+        }
+    }
+    assert DeepDiff(path_store.path_store,
+                    expected_path_store,
+                    ignore_order=True) == {}
+
+  @patch('server.lib.fetch.triples')
+  def test_traversal_without_sampling(self, mock_fetch):
+    mock_fetch.side_effect = fetch_triples_side_effect
+
+    def null_sample(_):
```

Comment:
curious why using this instead of using an input which will cause sampling to not happen OR if sampling always happens then we don't need this test case

Reply:
this is to indicate that the focus of this test is on the functionality of traversal and that the sampling logic is tested elsewhere in the file. Otherwise there would be more mocking and more convoluted inputs/outputs. 

I can add the extra mocks for the sampling part if you think it's best to do so! Let me know!